sbt-plugin/src/main/scala/ch/epfl/scala/AnalyzeDependencyGraph.scala

```diff
@@ -0,0 +1,220 @@
+package ch.epfl.scala
+
+import java.nio.file.Paths
+
+import scala.concurrent.Await
+import scala.concurrent.duration.Duration
+import scala.sys.process._
+import scala.util.Properties
+import scala.util.Try
+
+import ch.epfl.scala.GithubDependencyGraphPlugin.autoImport._
+import ch.epfl.scala.githubapi._
+import gigahorse.FullResponse
+import gigahorse.HttpClient
+import gigahorse.support.asynchttpclient.Gigahorse
+import sbt._
+import sbt.internal.util.complete._
+import sjsonnew.shaded.scalajson.ast.unsafe.JArray
+import sjsonnew.shaded.scalajson.ast.unsafe.JField
+import sjsonnew.shaded.scalajson.ast.unsafe.JObject
+import sjsonnew.shaded.scalajson.ast.unsafe.JString
+import sjsonnew.support.scalajson.unsafe.{Parser => JsonParser}
+
+object AnalyzeDependencyGraph {
+
+  object Model {
+    val help =
+      "download and display CVEs alerts from Github, and analyze them against dependencies (use hub or gh local config or GIT_TOKEN env var to authenticate, requires githubGenerateSnapshot)"
+
+    def blue(str: String): String = s"\u001b[34m${str}\u001b[0m"
+
+    case class Vulnerability(
+        packageId: String,
+        vulnerableVersionRange: String,
+        firstPatchedVersion: String,
+        severity: String
+    ) {
+      def severityColor: String = severity match {
+        case "critical" => "\u001b[31m"
+        case "high"     => "\u001b[31m"
+        case "medium"   => "\u001b[33m"
+        case "low"      => "\u001b[32m"
+        case _          => "\u001b[0m"
+      }
+
+      def coloredSeverity: String = s"${severityColor}${severity}\u001b[0m"
+
+      override def toString: String =
+        s"${blue(packageId)} [ $vulnerableVersionRange ] fixed: $firstPatchedVersion $coloredSeverity"
+    }
+
+    case class AnalysisParams(arg: Option[String])
+
+    sealed trait Vulnerable
+    object Good extends Vulnerable
+    object Bad extends Vulnerable
+    object No extends Vulnerable
+  }
+
+  import Model._
+
+  val AnalyzeDependencies = "githubAnalyzeDependencies"
+  private val AnalyzeDependenciesUsage =
+    s"""$AnalyzeDependencies [pattern]"""
+  private val AnalyzeDependenciesDetail = s"""Analyze the dependencies based on a search pattern:
+  $help
+  """
+
+  val commands: Seq[Command] = Seq(
+    Command(AnalyzeDependencies, (AnalyzeDependenciesUsage, AnalyzeDependenciesDetail), AnalyzeDependenciesDetail)(
+      extractPattern
+    )(analyzeDependencies)
+  )
+
+  private lazy val http: HttpClient = Gigahorse.http(Gigahorse.config)
+
+  private def extractPattern(state: State): Parser[AnalysisParams] =
```

Comment:
Maybe the name of this def should be `parser`.

```suggestion
  private def parser(state: State): Parser[AnalysisParams] =
```